### PR TITLE
Gcp folder tag

### PR DESCRIPTION
**Pull request by johnnywilkes**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/EXPANDR-3490

## Description
**GCP-IAM Pack**
Added the **gcp-iam-tagbindings-list** command to the integration to pull tags that are on project/folder/organization items.  
image

**GCP-Enrichment-Remediaton Pack** 
Added **GCPProjectHierarchy** script that find the parent container (folder/organization) of a GCP project all the way up to the organization object.

Incorporates the **GCPProjectHierarchy** script as well as **gcp-iam-tagbindings-list** command into playbook **GCP - Enrichment**
image

**Cortex ASM Pack** 
**Cortex ASM - ASM Alert**
- Updated the playbook to include the new *BypassDevCheck* input and added logic if it is set to true to allow remediation automation in cases that production workloads are possibly run on an endpoint.
- Updated the playbook to include the new *RemediationNotificationSubject* and *RemediationNotificationHTMLBody* inputs and so that email sent post automated remediation is now customizable.
image
image
Updated the **Cortex ASM - Remediation Path Rules** playbook to include the new *BypassDevCheck* input and added logic if it is set to true to allow remediation automation in cases that production workloads are possibly run on an endpoint.
image
image
Updated the **Cortex ASM - GCP Enrichment** playbook to update the "ASM - Tags" field with tags pulled from project/folder/organization objects.
image

_We have a large customer that deems this as important._

## Screenshots
above

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@samuelFain , can you add @efelmandar or another member of the playbook team, please?
@ShirleyDenkberg , can you please review?

**Comment by johnnywilkes** [comment]
@samuelFain , I believe I shared the demo video with you.  I shared with @efelmandar  as well.

File: Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.yml [hunk 1]
```diff
@@ -1561,8 +1453,7 @@ script:
     name: gcp-iam-organization-role-permission-remove
   - arguments:
     - default: false
-      description: The name of the organization that contains the custom role. For
-        example organizations/12345.
+      description: 'The name of the organization that contains the custom role. For example organizations/12345.'
```
Comment (ShirleyDenkberg) [inline Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.yml:1456]:
```suggestion
      description: 'The name of the organization that contains the custom role. For example: organizations/12345.'
```

File: Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.yml [hunk 2]
```diff
@@ -845,24 +785,21 @@ script:
...
+      description: The resource name of the entity under which this group resides in the Cloud Identity resource hierarchy.
       type: String
     - contextPath: GCPIAM.Group.updateTime
       description: The most recent time the group was modified.
       type: Date
   - arguments:
     - default: false
-      description: The parent resource of the groups to retrieve. This parameter is
-        usually equal to the organization customer ID. For example customers/C01234.
+      description: The parent resource of the groups to retrieve. This parameter is usually equal to the organization customer ID. For example customers/C01234.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.yml:795]:
```suggestion
      description: 'The parent resource of the groups to retrieve. This parameter is usually equal to the organization customer ID. For example: customers/C01234.'
```

File: Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.yml [hunk 3]
```diff
@@ -1409,16 +1311,13 @@ script:
     name: gcp-iam-service-account-key-delete
   - arguments:
     - default: false
-      description: The name of the organization that contains the custom role. For
-        example organizations/1234567.
+      description: The name of the organization that contains the custom role. For example organizations/1234567.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.yml:1314]:
```suggestion
      description: 'The name of the organization that contains the custom role. For example: organizations/1234567'.
```

File: Packs/GCP-IAM/ReleaseNotes/1_0_16.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### GCP-IAM
+- Added the **gcp-iam-tagbindings-list** command.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-IAM/ReleaseNotes/1_0_16.md:4]:
```suggestion
- Added the ***gcp-iam-tagbindings-list*** command.
```

**Comment by ShirleyDenkberg** [comment]
@samuelFain Doc review completed.

**Comment by johnnywilkes** [comment]
@ShirleyDenkberg , thanks for your review!

**Comment by johnnywilkes** [comment]
@BigEasyJ / @capanw , this is the PR we talked about yesterday.  If you can take care of it while I am away, please and thank you.

**Comment by samuelFain** [comment]
Hi @johnnywilkes and thank you for your contribution!
I will start my review shortly (got the demo video as well).
I added @efelmandar as a reviewer as well.

**Review by samuelFain — COMMENTED** [review]
@BigEasyJ / @capanw / @johnnywilkes, Awesome job!
Added a few minor comments, @efelmandar will also review this PR shortly.
I'm available for any questions in the meanwhile.

BTY, @johnnywilkes thanks for the demo video it was super helpful!

File: Packs/GCP-Enrichment-Remediation/Scripts/GCPProjectHierarchy/GCPProjectHierarchy.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    """
+    temp: Dict[str, str] = {}
+    if "folder" in parent_obj:
+        folder_info = execute_command("gcp-iam-folders-get", {"folder_name": parent_obj})
+        if not folder_info:
+            return "NONE", temp
+        name = "folders/" + folder_info['displayName']
+        temp["level"] = str(level)
+        temp["id"] = name
+        temp["number"] = folder_info['name']
```
[the remark this answers is not in the document]
Comment (BigEasyJ) [reply Packs/GCP-Enrichment-Remediation/Scripts/GCPProjectHierarchy/GCPProjectHierarchy.py:31]: Does this suggestion apply for Line 28 as well? 

Also, do we prefer `get` over square bracket notation? I think the square bracket notation should be fine here and intends to raise exceptions if needed for non existing keys.

[the remark this answers is not in the document]
Comment (capanw) [reply Packs/GCP-Enrichment-Remediation/Scripts/GCPProjectHierarchy/GCPProjectHierarchy.py:31]: Agreed with @BigEasyJ. get doesn't throw a keyError.

**Review by efelmandar — APPROVED** [review]

File: Packs/GCP-Enrichment-Remediation/Scripts/GCPProjectHierarchy/GCPProjectHierarchy.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (samuelFain) [reply Packs/GCP-Enrichment-Remediation/Scripts/GCPProjectHierarchy/GCPProjectHierarchy.py:31]: Got it, as long as you aware of the possible keyError and handle it, I accept your comment.

**Comment by samuelFain** [comment]
@capanw, following up on your conversation with @ShacharKidor, please update this branch from master so we can resolve the conflicts and merge. thanks!

**Pull request merged**